### ISSUE TITLE
SortedList Slice Assignment Differs Slightly With Python list

### ISSUE THREAD

**Issue by agroce** [grantjenks/sorted_containers#61]
Random testing again:

```
>>> l1 = SortedList(['A'])
>>> l1[1:0] = ['B']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/sortedcontainers/sortedlist.py", line 771, in __setitem__
    raise ValueError(message)
ValueError: 'B' not in sort order at index 0
>>> l2 = ['A']
>>> l2[1:0] = ['B']
>>> l2
['A', 'B']

```

**Comment by agroce** [comment]
Note:

`>>> l = SortedList(['B'])
>>> l[1:0] = ['A']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/sortedcontainers/sortedlist.py", line 766, in __setitem__
    raise ValueError(message)
ValueError: 'A' not in sort order at index 1
`

**Comment by grantjenks** [comment]
Bug is at [sortedlist.py:768](https://github.com/grantjenks/sorted_containers/blob/master/sortedcontainers/sortedlist.py#L768) The check for stop is meaningless because the slice is empty. The fast-path case (step = 1) should assume stop is greater than start.

Strange case. Thanks for finding it. I'll have a fix out in a few days.

**Comment by grantjenks** [comment]
Fixed at 501d4a9 and deployed at 1.5.7
